### PR TITLE
@sweir27 => Implement responsive auction/:id views

### PR DESCRIPTION
**Pull request by damassi**
This PR serves a few purposes: 

- Set down a long-term pattern for implementing React-based server-side rendering in Force
- Rewrite existing Jade components into React components
- Implement responsive mobile views for `/auction/:id`
- Modularize / refactor `<ArtworkBrowser />` component in preparation for new `/auctions` designs
- BEMify a lot of existing CSS
- Expand test coverage
- Refactor Redux-based components to move logic into selectors, as well as centralize all subapp state in reducers. 

See:  https://github.com/artsy/auctions/projects/1

**TODO:**
- Wider device testing
- Polish

**Example:** 
- http://localhost:5000/auction2/phillips-summer-school 
- Open Chome dev tools and click the mobile responsive icon, refresh page 

screen shot 2017-07-20 at 5 56 29 pm

--- 

screen shot 2017-07-20 at 5 56 54 pm

---

screen shot 2017-07-20 at 6 03 58 pm

--- 

screen shot 2017-07-20 at 5 57 54 pm

--- 

screen shot 2017-07-20 at 5 58 36 pm

--- 

screen shot 2017-07-20 at 5 59 39 pm

### CODE REVIEW

File: lib/middleware/error_handler.coffee [hunk 1]
```diff
@@ -7,6 +7,7 @@ module.exports = (err, req, res, next) ->
     '../../desktop/components/error_handler/index.jade'
   )
   isNotProduction = NODE_ENV isnt 'production'
+  console.log err if isNotProduction
```
Comment (damassi) [inline lib/middleware/error_handler.coffee]: Fixes issue where stylus compile-time errors don't report in console and dev has no clue whats going on

File: desktop/apps/auction2/components/DOM.js [hunk 2]
```diff
@@ -10,19 +10,19 @@ class DOM extends Component {
...
-  }
-
   // Selectors
   $ = null
   $body = null
   $registerBtn = null
 
   componentDidMount () {
+    const FastClick = require('fastclick')
+    FastClick(document.body)
```
Comment (damassi) [inline desktop/apps/auction2/components/DOM.js]: @sweir27 and @erikdstock - When building responsive views make sure to use this; it removes the weird 300ms delay from clicks / taps

Comment (sweir27) [inline desktop/apps/auction2/components/DOM.js]: 👍 cool, didn't know about this.

File: desktop/apps/auction2/components/Layout.js [hunk 3]
```diff
@@ -39,7 +39,7 @@ function Layout (props) {
           <div className='auction2-main-page'>
             { showFollowedArtistsRail &&
               <WorksByFollowedArtists /> }
-            <CommercialFilter />
+            <ArtworkBrowser />
```
Comment (damassi) [inline desktop/apps/auction2/components/Layout.js]: This is going to soon be shared with `/auctions` so restructured slightly to make more portable

File: lib/middleware/error_handler.coffee (hunk 1 above)
Comment (erikdstock) [inline lib/middleware/error_handler.coffee]: `isnt`? 😵

**Comment by erikdstock** [comment]
Cool! Playing with it a little this morning while I sit at a coffee shop. I'm using safari on the ios simulator, so this is all mobile only. I tried using the different sorts, expanding the hamburger menu and sorting again, going to the info page, etc.
- got an error for needing to be admin when i tried to visit the auction url above; after i logged in I was able to get there. When I logged out I got the error again.
image

```
Force listening on port 5000
GET /assets/app.js - - ms - -
{ Error: You must be logged in as an admin
    at module.exports (/Users/eriks/dev/artsy/force/desktop/lib/admin_only.coffee:5:13)
    at Layer.handle [as handle_request] (/Users/eriks/dev/artsy/force/node_modules/express/lib/router/layer.js:95:5)
    at next (/Users/eriks/dev/artsy/force/node_modules/express/lib/router/route.js:137:13)
    at Route.dispatch (/Users/eriks/dev/artsy/force/node_modules/express/lib/router/route.js:112:3)
    at Layer.handle [as handle_request] (/Users/eriks/dev/artsy/force/node_modules/express/lib/router/layer.js:95:5)
    at /Users/eriks/dev/artsy/force/node_modules/express/lib/router/index.js:281:22
    at param (/Users/eriks/dev/artsy/force/node_modules/express/lib/router/index.js:354:14)
    at param (/Users/eriks/dev/artsy/force/node_modules/express/lib/router/index.js:365:14)
    at Function.process_params (/Users/eriks/dev/artsy/force/node_modules/express/lib/router/index.js:410:3)
    at next (/Users/eriks/dev/artsy/force/node_modules/express/lib/router/index.js:275:10)
    at Function.handle (/Users/eriks/dev/artsy/force/node_modules/express/lib/router/index.js:174:3)
    at router (/Users/eriks/dev/artsy/force/node_modules/express/lib/router/index.js:47:12)
    at /Users/eriks/dev/artsy/force/desktop/apps/auction2/index.js:12:5
```
-----

- Kind of weird extra arrow here:
image
-----

- Auction info page garbled:
image
- Likewise here (maybe same issue) the footer below all the artworks is all big:
image


-----

See screencap below for these two:
- The LAMA 'bidding open' notification is a little conspicuous- do we still want it? How do we handle other responsive views with this? I guess it looks fine besides the way that it doesn't know where to go when you scroll.
- There's an empty box at the top- I *assume* that is due to my local env and that it would normally have the 'download artsy app' CTA but want to be sure.
auctionscroll

File: desktop/apps/auction2/components/artwork_browser/header/FilterSort.js [hunk 4]
```diff
@@ -16,9 +17,11 @@ function FilterSort (props) {
   const itemIndex = Object.keys(sortMap).indexOf(selectedSort)
   const optionsOffset = (itemIndex) * itemHeight
 
+  const b = block('auction2-filter-sort')
```
Comment (sweir27) [inline desktop/apps/auction2/components/artwork_browser/header/FilterSort.js]: 👏

File: desktop/apps/auction2/components/artwork_browser/header/HeaderDesktop.js [hunk 5]
```diff
@@ -52,17 +42,27 @@ function Header (props) {
...
 const mapStateToProps = (state) => {
+  const {
+    isFetchingArtworks,
+    isListView,
+    total
+  } = state.artworkBrowser
+
+  const displayType = isListView ? 'list' : 'grid'
+
+  const totalLabel = isFetchingArtworks
```
Comment (sweir27) [inline desktop/apps/auction2/components/artwork_browser/header/HeaderDesktop.js]: 👏  Big fan of this pattern (keeping selectors out of the components)

**Comment by sweir27** [comment]
Also re: @erikdstock's testing, I am pretty sure we should _not_ be rendering the auction reminders component in mobile web at all, but @devangt / @katarinabatina can verify.

**Comment by damassi** [comment]
I confirmed with @katarinabatina earlier today -- no auction reminders.

**Comment by damassi** [comment]
@sweir27 and @erikdstock - I'm going to go ahead and merge this since its still behind the adminOnly path; would like to test on a few more devices in staging.

**Pull request merged**